xcode/Subconscious/SubconsciousTests/Tests_NoosphereService.swift

```diff
@@ -21,7 +21,34 @@ final class Tests_NoosphereService: XCTestCase {
         )
         return url
     }
+    
+    func testSphereTraversal() throws {
+        let tmp = try TestUtilities.createTmpDir()
+        let data = try TestUtilities.createDataService(tmp: tmp)
+        let noosphere = data.noosphere
+        
+        let bobReceipt = try noosphere.createSphere(ownerKeyName: "bob")
+        let aliceReceipt = try noosphere.createSphere(ownerKeyName: "alice")
+        
+        let aliceSphere = try noosphere.open(identity: aliceReceipt.identity)
+        let bobSphere = try noosphere.open(identity: bobReceipt.identity)
+        
+        try bobSphere.setPetname(did: aliceReceipt.identity, petname: "alice")
+        try aliceSphere.setPetname(did: bobReceipt.identity, petname: "bob")
+        try noosphere.save()
+        
+        let aliceDid = try bobSphere.getPetname(petname: "alice")
+        XCTAssertEqual(aliceDid, aliceReceipt.identity)
+        
+//        let result = try noosphere.sync() 
+        let newSphere = try noosphere.traverse(sphere: bobSphere, petname: "alice")
```

Comment:
Am I right in thinking I need to actually `sync` before this will work?

Reply:
Bob's sphere records Alice's sphere's identity in Bob's address book, but a gateway is needed to resolve the identity into a Link (CID) -- some ideas:

* Compartmentalize some tests that require a running gateway (integration tests), and run a gateway locally as part of the test (is `orb` a part of the build step anywhere?)
* Expose a way (possibly in FFI) to manually set the Link of a sphere, though the hydration would still require IPFS (or seeding into a store via side-channel)

We have a `test_kubo` feature in Noosphere that require running ipfs-desktop in conjunction for tests that require full IPFS node running when testing e.g. the gateway/name system. I suspect there'll be some heavier workflows worth testing in this way